rules/do-you-use-loggermessage-in-net.md/rule.md

```diff
@@ -0,0 +1,114 @@
+
+```csharp
+private static readonly Action<ILogger, WorkItem, Exception> s_processingPriorityItem =
+    LoggerMessage.Define<WorkItem>(
+        LogLevel.Information,
+        new EventId(1, nameof(PriorityItemProcessed)),
+        "Processing priority item: {Item}");
+
+private static readonly Action<ILogger, Exception> s_failedToProcessWorkItem = LoggerMessage.Define<Exception>(
+    LogLevel.Critical,
+    new EventId(13, nameof(FailedToProcessWorkItem)),
+    "Epic failure processing item!");
+```
+
+This delegate is then invoked using strongly typed extension methods like PriorityItemProcessed, making it simple to log information in a structured manner with minimal performance impact.
+
+```csharp
+public static void PriorityItemProcessed(
+    this ILogger logger, WorkItem workItem) =>
+    s_processingPriorityItem(logger, workItem, default!);
+
+public static void FailedToProcessWorkItem(
+    this ILogger logger, Exception  ex) =>
+    s_failedToProcessWorkItem(logger, ex);
+```
+
+Usage Example:
+
+```csharp
+protected override async Task ExecuteAsync(
+    CancellationToken stoppingToken)
+{
+    using (IDisposable? scope = logger.ProcessingWorkScope(DateTime.Now))
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            try
+            {
+                WorkItem? nextItem = priorityQueue.ProcessNextHighestPriority();
+
+                if (nextItem is not null)
+                {
+                    logger.PriorityItemProcessed(nextItem);
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.FailedToProcessWorkItem(ex);
+            }
+
+            await Task.Delay(1_000, stoppingToken);
+        }
+    }
+}
+
+```
+
+## Optimizing Further with ILogger.IsEnabled
+
+For even better performance, you can check whether logging is enabled for a particular log level using `ILogger.IsEnabled(LogLevel)`. This check ensures that logging statements aren't called when the log level is not configured, thus avoiding unnecessary object allocations and improving efficiency.
+
+```csharp
+if (logger.IsEnabled(LogLevel.Information))
+{
+    logger.Information("This is a log message.");
+}
+```
+
+## When to Use LoggerMessage
+
+The LoggerMessage pattern is particularly beneficial in scenarios where logging needs to be performed frequently with minimal performance impact. It is most effective when:
+
+- Logging is done in high-throughput scenarios (e.g., background services, real-time applications).
+- You require structured logging for easier integration with advanced logging systems (e.g., Serilog, ElasticSearch).
+- Reducing object allocations and computational overhead is a priority for the application.
+
+## More information on LoggerMessage
+
+There is a great article on Microsoft Learn which goes into more detail and usage examples: <https://learn.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging>
```

Comment:
Hey @tomek-i 

- [ ] 1. Can you try making this link descriptive (e.g. grab the title from the article)